sled-agent/src/rack_setup/plan/service.rs

```diff
@@ -28,6 +28,9 @@ use std::path::{Path, PathBuf};
 use thiserror::Error;
 use uuid::Uuid;
 
+// The number of boundary NTP servers to create from RSS.
+const BOUNDARY_NTP_COUNT: usize = 2;
```

Comment:
We *might* want to make this `1` for now, and have it be a configuration option. Totally agree with this disposition on the production build, but it might make development builds painful.

If we require a minimum of `2`, will it be possible to call RSS when running from a single test sled?

Reply:
As it is now, two is okay here even with a single sled. The first BOUNDARY_NTP_COUNT sleds get their NTP server configured in a boundary role, all others get it set up as an internal server. If there is just one sled, we end up with one boundary server.

In a change from an earlier version of this PR, having now decoupled the boundary server role from the scrimlets, the NTP servers don't start until after RSS in the bootstrap case. The sequence is (based on your earlier comment):

1. Boot to bootstrap network
1. Operator provides underlay address, DNS + NTP configuration through wicket / RSS
1. RSS starts sled agents, tells them to use underlay
1. RSS starts Internal DNS and NTP
1. Wait for all sleds to have synchronised time (that's the second commit in this branch that I've been testing today).
1. RSS starts all other services

Internal servers find the boundary ones via internal DNS, boundary ones use the settings from the operator.

I still have to test the cold reboot case.

Reply:
Does the _operator_ provide the underlay addresses or do we just generate those ourselves?

Reply:
They're generated by RSS and persisted in the service plan.

```toml
[[services."[fd00:1122:3344:101::1]:12345".service]]
addresses = ["fd00:1122:3344:101::10"]
gz_addresses = []
id = "3db6e379-17e4-48da-b6f5-34b80c114343"
zone_type = "ntp"

[[services."[fd00:1122:3344:101::1]:12345".service.services]]
boundary = true
servers = ["0.pool.ntp.org"]
type = "ntp"

[[services."[fd00:1122:3344:101::1]:12345".service.services]]
servers = ["9.9.9.9", "80.80.80.80"]
type = "dns_client"
```

along with the DNS entries that are used by the non-boundary NTP servers
(this could be switched to just use the IPs directly too, and that might be worthwhile).

```
[[dns_config.zones.records.records]]
data = "fd00:1122:3344:101::10"
type = "AAAA"

[dns_config.zones.records.key]
name = "3db6e379-17e4-48da-b6f5-34b80c114343.host"
```